server/eclipse-project/src/main/java/us/freeandfair/corla/report/ReportRows.java

```diff
@@ -0,0 +1,435 @@
+    row.put("re-audit ballot comment", acvr.getComment());
+    return row;
+  }
+
+  /** add fields unique to results report **/
+  public static Row addResultsFields(final Row row, final Tribute tribute, final Integer multiplicity) {
+    row.put("multiplicity", toString(multiplicity));
+    return addResultsFields(row, tribute);
+  }
+
+  /** add fields unique to activity report, if the multiplicity is unknown **/
+  public static Row addResultsFields(final Row row, final Tribute tribute) {
+    row.put("random number", toString(tribute.rand));
+    row.put("random number sequence position", toString(tribute.randSequencePosition));
+    return row;
+  }
+
+  /** tie the headers to a row **/
+  public static class ActivityReport {
+    /** a selection of headers **/
+    public static final String[] HEADERS =
+      ArrayUtils.removeElements(ArrayUtils.clone(ALL_HEADERS),
+                                "random number sequence position",
+                                "random number",
+                                "multiplicity");
+
+    /** no instantiation **/
+    private ActivityReport() {};
+
+    /** new row **/
+    public static final Row newRow() {
+      return new Row(HEADERS);
+    }
+  }
+
+  /** tie the headers to a row **/
+  public static class ResultsReport {
+    /** a selection of headers **/
+    public static final String[] HEADERS =
+      ArrayUtils.removeElements(ArrayUtils.clone(ALL_HEADERS),
+                                "revision",
+                                "re-audit ballot comment");
+
+    /** no instantiation **/
+    private ResultsReport() {};
+
+    /** new row **/
+    public static final Row newRow() {
+      return new Row(HEADERS);
+    }
+  }
+
+  /** tie the headers to a row **/
+  public static class SummaryReport {
+
+    /** a selection of headers **/
+    public static final String[] HEADERS = {
+      "Contest",
+      "targeted",
+      "Winner",
+
+      "Risk Limit Achieved",
+      "diluted margin",
+      "disc +2",
+      "disc +1",
+      "disc -1",
+      "disc -2",
+      "gamma",
+
+      "ballot count",
+      "min margin",
+      "votes for winner",
+      "votes for runner up",
+      "total votes (marked)",
+      "disagreement count (included in +2 and +1)"
+    };
+
+    /** no instantiation **/
+    private SummaryReport() {};
+
+    /** new Row **/
+    public static final Row newRow() {
+      return new Row(HEADERS);
+    }
+  }
+
+  /** risk limit achieved according to math.Audit **/
+  public static String riskLimitAchieved(final ComparisonAudit ca) {
+    final BigDecimal result =  Audit.pValueApproximation(ca.getAuditedSampleCount(),
+                                                         ca.getDilutedMargin(),
+                                                         ca.getGamma(),
+                                                         ca.discrepancyCount(-2),
+                                                         ca.discrepancyCount(-1),
```

Comment:
The "unders" need to be flipped around, the order accepted by `pValueApproximation` is `oneUnder`, `twoUnder`, `oneOver`, `twoOver`. That's what the note over in the math package is about (this differs from calculating the optimistic samples to audit). I realize whoever ordered those arguments was thinking about the internal representation of the discrepancy count as it would appear on a number line (and thus how the call site would read when called exactly like this, but only like this), but a reader of the method's argument list would find it strangely asymmetric.